rust-engine/src/ast.rs

```diff
@@ -1487,8 +1487,26 @@ pub mod traits {
         };
     }
 
-    derive_has_metadata!(Item, Expr, Pat);
-    derive_has_kind!(Item => ItemKind, Expr => ExprKind, Pat => PatKind);
+    derive_has_metadata!(
+        Item,
+        Expr,
+        Pat,
+        Guard,
+        Arm,
+        ImplItem,
+        TraitItem,
+        GenericParam
+    );
+    derive_has_kind!(
+        Item => ItemKind, Expr => ExprKind, Pat => PatKind, Guard => GuardKind,
+        GenericParam => GenericParamKind, ImplItem => ImplItemKind, TraitItem => TraitItemKind
+    );
+
+    impl HasSpan for Attribute {
+        fn span(&self) -> Span {
+            self.span.clone()
+        }
+    }
```

Comment:
Is there a reason the trait method returns an owned span over a reference?

Reply:
Ah, good catch: indeed that looks wierd right now!
It's returning an owned span because we want to intern spans and make them `Copy`able: right now we are cloning spans everywhere for this reason.
Soon, we'll have `Span(&'static InnerSpan)` or `Span {index: usize}`, so we won't have that issue any longer, spans will always be owned.